maanika/sound-map
# Add Battery Level Monitoring

## Description

This PR adds the functionality of monitoring the battery level and vocalizing the battery percentage every 10 seconds.

## Screenshots
The below images shows as example of the text output from the program..
<img width="555" alt="Capture" src="https://user-images.githubusercontent.com/55540980/108586131-8f3c7e00-73a0-11eb-8a20-2cadc292e3ab.PNG">

The below image shows the hardware setup.
<img width="491" alt="Capture" src="https://user-images.githubusercontent.com/55540980/108605946-4d94ed00-740b-11eb-8964-80d8eb16370b.PNG">

The below file shows the results obtained while testing the program on campus. The GPS outputs are removed to make the file easier to follow. 
[Program_Output.txt](https://github.com/maanika/sound-map/files/6016029/Program_Output.txt)
The user starts at Campbell hall and then selects the destination as Campbell hall. Next the user selects Campus center as the destination. The user follows the checkpoint output and arrives at Campus center. Finally the user selects Hargrave library as the destination.
Comments have been added in the file alongside the output of the program.
The battery voltage throughout the journey was recorded in between 7.56V and 7.72V, which corresponds to 60%.

## Steps to Test

1. Download the code and rename the project folder appropriately.
2. Observe the output strings through an UART connection.
3. Observe the speech of the battery level every ten seconds through the earphones.
